README.md

```diff
@@ -89,13 +94,48 @@ the library file `libmysqlclient.so` but is missing the header file `mysql.h`
 
 ### Mac OS X
 
-You may use MacPorts, Homebrew, or a native MySQL installer package. The most
+You may use Homebew, MacPorts, or a native MySQL installer package. The most
 common paths will be automatically searched. If you want to select a specific
 MySQL directory, use the `--with-mysql-dir` or `--with-mysql-config` options above.
 
 If you have not done so already, you will need to install the XCode select tools by running
 `xcode-select --install`.
 
+Later versions of MacOS no longer distribute a linkable OpenSSL library. It is
+common to use Homebrew or MacPorts to install OpenSSL. Make sure that both the
+Ruby runtime and MySQL client libraries are compiled with the same OpenSSL
+family, 1.0 or 1.1 or 3.0, since only one can be loaded at runtime.
+
+``` sh
+$ brew install openssl@1.1
+$ gem install mysql2 -- --with-openssl-dir=$(brew --prefix openssl@1.1)
+
+or
+
+$ sudo port install openssl11
+```
+
+Since most Ruby projects use Bundler, you can set build options in the Bundler
+config rather than manually installing a global mysql2 gem. This example shows
+how to set Bundler build arguments:
+
+``` sh
+$ bundle config --local build.mysql2 -- --with-openssl-dir=$(brew --prefix openssl@1.1)
+```
+
+Another helpful trick is to use the same OpenSSL library that your Ruby was
+built with. This example finds the argument `--with-openssl-dir=/some/path`
+from the Ruby build and adds that to the Bundler config:
```

Comment:
I think it's helpful to explain this example works only "if the Ruby is configured with the `--with-openssl-dir=/some/path`". I tried the example below for my built Ruby on local, and I couldn't find the `--with-openssl-dir=/some/path`, because maybe the Ruby was not configured with `--with-openssl-dir=/some/path`.

```
$ which ruby
/usr/local/ruby-3.2.0-preview2/bin/ruby

$ ruby -r rbconfig -e 'puts RbConfig::CONFIG["configure_args"]' | xargs -n1
--prefix=/usr/local/ruby-3.2.0-preview2
--enable-shared
```